velox/functions/prestosql/SIMDJsonFunctions.h

```diff
@@ -0,0 +1,253 @@
+  call(bool& result, const arg_type<Json>& json, const TInput& value) {
+
+    std::string jsonData(json);
+    ParserContext ctx(jsonData.data(), jsonData.length());
+    std::string jsonpath = "";
+    result = false;
+
+    do {
+      try {
+        ctx.parseDocument();
+      } catch (simdjson::simdjson_error& e) {
+        printf(
+            "error: Failed to parse json as document. error :%s\n",
+            simdjson::error_message(e.error()));
+        break;
+      }
+
+      if (ctx.jsonDoc.type() != simdjson::ondemand::json_type::array) {
+        break;
+      }
+
+      try {
+        for (auto&& v : ctx.jsonDoc) {
+          if constexpr (std::is_same_v<TInput, bool>) {
+            if (v.type() == simdjson::ondemand::json_type::boolean &&
+                v.get_bool() == value) {
+              result = true;
+              break;
+            }
+          } else if constexpr (std::is_same_v<TInput, int64_t>) {
+            if (v.type() == simdjson::ondemand::json_type::number &&
+                ((v.get_number_type() ==
+                      simdjson::ondemand::number_type::signed_integer &&
+                  v.get_int64() == value) ||
+                 (v.get_number_type() ==
+                      simdjson::ondemand::number_type::unsigned_integer &&
+                  v.get_uint64() == value))) {
+              result = true;
+              break;
+            }
+          } else if constexpr (std::is_same_v<TInput, double>) {
+            if (v.type() == simdjson::ondemand::json_type::number &&
+                v.get_number_type() ==
+                    simdjson::ondemand::number_type::floating_point_number &&
+                v.get_double() == value) {
+              result = true;
+              break;
+            }
+          } else {
+            if (v.type() == simdjson::ondemand::json_type::string) {
+              std::string_view rlt = v.get_string();
+              std::string str_value = value.getString();
+              if (rlt.compare(str_value) == 0) {
+                result = true;
+                break;
+              }
+            }
+          }
+        }
+      } catch (simdjson::simdjson_error& e) {
+        if (e.error() != simdjson::INCORRECT_TYPE &&
+            e.error() != simdjson::NUMBER_ERROR) {
+        }
+      }
+    } while (0);
+  }
+};
+
+template <typename T>
+struct SIMDJsonParseFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& json) {
+
+    std::string jsonData(json);
+    ParserContext ctx(jsonData.data(), jsonData.length());
+    bool retVal = false;
+
+    try{
+      ctx.parseElement();
+      std::string_view rlt_tmp = simdjson::to_string(ctx.jsonEle);
+      std::string rlt(rlt_tmp);
+      UDFOutputString::assign(result, rlt);
+      retVal = true;
+    } catch(simdjson::simdjson_error& e)
+    {
+      VELOX_USER_FAIL("Cannot convert '{}' to JSON", jsonData);
```

Comment:
jsonData can be very large. Perhaps, truncate to some reasonable limit before including in the error message and put it at the end of the message.

```
"Cannot parse JSON: {}, {}", e.what(), truncate(jsonData)
```